core/trino-spi/src/main/java/io/trino/spi/block/EncoderUtil.java

```diff
@@ -120,4 +134,156 @@ public static byte[] retrieveNullBits(SliceInput sliceInput, int positionCount)
             throw new UncheckedIOException(e);
         }
     }
+
+    static void compactBytesWithNullsVectorized(SliceOutput sliceOutput, byte[] values, boolean[] isNull, int offset, int length)
+    {
+        requireNonNull(sliceOutput, "sliceOutput is null");
+        checkFromIndexSize(offset, length, values.length);
+        checkFromIndexSize(offset, length, isNull.length);
+        byte[] compressed = new byte[length];
+        int valuesIndex = 0;
+        int compressedIndex = 0;
```

Comment:
Please rename all the "compressed" variables to "compacted" or similar

Reply:
The original implementation used compressed as the variable name so I just kept it, I can change all occurance of compressed to compacted.